apps/ejabberd/test/ejabberd_listener_SUITE.erl

```diff
@@ -0,0 +1,31 @@
+-module(ejabberd_listener_SUITE).
+
+-compile([export_all]).
+
+all() ->
+    [tcp_socket_is_started_with_options].
+
+
+tcp_socket_is_started_with_options(_C) ->
+
+    meck:new(gen_tcp, [unstick, passthrough]),
+    meck:expect(gen_tcp, listen, fun(Port, Opts) ->
+                                         meck:passthrough([Port, Opts])
+                                 end),
+
+    OverrideBacklog = {backlog, 50},
+    {ok, _Pid} = listener_started([OverrideBacklog]),
+
+    [{_Pid, {gen_tcp, listen, [_, Opts]}, _Result}] =  meck:history(gen_tcp),
+
+    DefaultBacklog = {backlog, 100},
+    [DefaultBacklog, OverrideBacklog] = proplists:lookup_all(backlog, Opts),
```

Comment:
Same as above: If the listening socket is started with `[{opt_a, "a"}, {opt_a, "b"}]`, then my guess is that it will take `"a"` as the value for `opt_a`. This test shows that the values are passed, but we can't be certain that the right value is used. I propose to do this the right way and **replace** the values in the original proplist if there are overrides.